app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -99,7 +99,16 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
                 class_name = "middleware_deployment";
                 break;
             case "MiddlewareServer":
-                class_name = "middleware_server";
+                switch (d.item.product) {
+                    case "Hawkular":
+                        class_name = "vendor-hawkular";
```

Comment:
thanks @pilhuhn . if those are different server types, why the product field was chosen to do the differentiation? Why aren't they different rails models with inheritance from the base middleware_server model? then the type field could serve as the differentiation. it could be easier to do queries on a specific type only, and they can still reside in the same db table.